### ISSUE TITLE
Should the mask provided in stereo fusion be undistorted?

### ISSUE THREAD

**Issue by f1shel** [colmap/colmap#1674]
If I use a camera model with distortion, say 'OPENCV', I must first undistort input images and then conduct dense reconstruction. The depth map and normal map reconstructed by patch-match are based on the undistorted resolution (typically several pixels smaller than the original image resolution). In the next step, we fuse the multiview depth map into a unified point cloud.
My question is, should the resolution/size of mask provided in stereo fusion match with the undistorted images? Or will the ImageReader in stereo_fusion do internal undistortion while fusing?

**Comment by f1shel** [comment]
https://github.com/colmap/colmap/blob/87b3aa325bd8e5fb913788e29e9ac1e085e28b67/src/mvs/fusion.cc#L342-L363

Mask image is read in `StereoFusion::InitFusedPixelMask` and `GpuMat::Read`. It simply reads the image content into an array-like object.

https://github.com/colmap/colmap/blob/87b3aa325bd8e5fb913788e29e9ac1e085e28b67/src/mvs/gpu_mat.h#L354-L376

Then colmap simply scales the mask and matchs its resolution to the depth image.

https://github.com/colmap/colmap/blob/87b3aa325bd8e5fb913788e29e9ac1e085e28b67/src/util/bitmap.cc#L571-L574

So, it seems that fusion masks are out of consideration of distortion. COLMAP assumes that masks in fusion process have already get rid of distortion, which is quite different with masks used in feature extraction.

Now, let's say I have masks for original distorted images. I can use them for feature extraction, but for mvs, I guess I need to undistort both images and masks?

**Comment by ahojnnes** [comment]
Yes, these are expected to be undistorted. It would be possible to "abuse" the image_undistorter for this task. You could simply run it on a folder with the masks, where the original (distorted) masks have the same image names as the actual images. Then use the output of this undistortion as the input for masking in the fusion.

**Comment by caligraf1** [comment]
Are there cmd lines for masks in dense reconstruction same as for feature_extraction? How to consider masks in mvs?
